hazelcast/src/test/java/com/hazelcast/quorum/atomic/AtomicLongQuorumWriteTest.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.quorum.atomic;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.core.IAtomicLong;
+import com.hazelcast.quorum.QuorumException;
+import com.hazelcast.quorum.QuorumType;
+import com.hazelcast.test.HazelcastParametersRunnerFactory;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import static com.hazelcast.quorum.QuorumType.READ_WRITE;
+import static com.hazelcast.quorum.QuorumType.WRITE;
+import static java.util.Arrays.asList;
+import static org.hamcrest.core.Is.isA;
+
+@RunWith(Parameterized.class)
+@Parameterized.UseParametersRunnerFactory(HazelcastParametersRunnerFactory.class)
+@Category({QuickTest.class})
+public class AtomicLongQuorumWriteTest extends AbstractAtomicQuorumTest {
+
+    @Parameterized.Parameter
+    public static QuorumType quorumType;
+
+    @Parameterized.Parameters(name = "quorumType:{0}")
+    public static Iterable<Object[]> parameters() {
+        return asList(new Object[][]{{WRITE}, {READ_WRITE}});
+    }
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @BeforeClass
+    public static void setUp() {
+        initTestEnvironment(new Config(), new TestHazelcastInstanceFactory());
+    }
+
+    @AfterClass
+    public static void tearDown() {
+        shutdownTestEnvironment();
+    }
+
+    @Test
+    public void addAndGet() throws Exception {
+        along(0).addAndGet(1);
```

Comment:
Is there a reason for not checking if the returned value is an expected one? This question valid for other tests too.

Reply:
The test is testing for quorum handling, not for data-structure correctness.